DEV-CONVENTIONS.md

```diff
@@ -46,43 +46,60 @@ Others docs you should look at:
 
 ## Logging conventions
 
-We use [glog](http://godoc.org/github.com/golang/glog) for internal logging,
-with the following conventions around log level choices:
-
-- glog.Errorf() - Always an error
-
-- glog.Warningf() - Something unexpected, but probably not an error
-
-- glog.Infof() has multiple levels:
-
-  - glog.V(0) - Generally useful for this to ALWAYS be visible to an operator
-    - Programmer errors
-    - Logging extra info about a panic
-    - CLI argument handling
-
-  - glog.V(1) - A reasonable default log level if you don't want verbosity.
-    - Information about config (listening on X, watching Y)
-    - Errors that repeat frequently that relate to conditions that can be corrected
-
-  - glog.V(2) - Useful steady state information about the service and important
-  log messages that may correlate to significant changes in the system.  This is
-  the recommended default log level for most systems.
-    - Logging HTTP requests and their exit code
-    - System state changing
-
-  - glog.V(3) - Extended information about changes
-    - More info about system state changes
-
-  - glog.V(4) - Debug level verbosity (for now)
-    - Logging in particularly thorny parts of code where you may want to come
-    back later and check it
-
-As per the comments, the practical production level at runtime is V(2). Developers and QE
-environments may wish to run at V(3) or V(4). If you wish to change the log
-level, you can pass in `-v=X` where X is the desired maximum level to log.
+Istio code is expected to use the standard [Istio logging package](https://godoc.org/istio.io/istio/pkg/log).
+When importing external package to Istio, if these packages send logs to the standard Go "log" package or
+to the global [zap logger](https://godoc.org/go.uber.org/zap), the log data will automatically be captured
+by the Istio log package and merged into the primary log stream(s) produced for the component. If
+other logging packages are used, their output will uncoordinated and unrelated to the primary log
```

Comment:
This seems like a major disadvantage of this change. I don't get to make choices for logging used in my libraries.

Reply:
?

Libraries can use any logging package they want. There's no difference with this new approach. However, if you use go "log" or zap, the logs your package emits will be captured by the Istio logger and rolled into the main log files and participate in log rotation, etc.

If you use a random library for logging, no behavior will change. If you use one of the ones we recommend, the user will get a better experience.

If there are other popular libraries that we want to capture the output, we can augment the Istio package to do so.

glog unforfunately doesn't support having its output captured.